snuba/datasets/entities/factory.py

```diff
@@ -0,0 +1,63 @@
+from typing import Callable, MutableMapping
+
+from snuba import settings
+from snuba.datasets.entity import Entity
+from snuba.datasets.entities import EntityKey
+from snuba.util import with_span
+
+ENTITY_IMPL: MutableMapping[str, Entity] = {}
+ENTITY_NAME_LOOKUP: MutableMapping[Entity, str] = {}
+
+
+class InvalidEntityError(Exception):
+    """Exception raised on invalid entity access."""
+
+
+@with_span()
+def get_entity(name: EntityKey) -> Entity:
+    if name in ENTITY_IMPL:
+        return ENTITY_IMPL[name]
+
+    if name in settings.DISABLED_ENTITIES:
+        raise InvalidEntityError(
+            f"entity {name!r} is not available in this environment"
+        )
+
+    from snuba.datasets.cdc.groupassignee_entity import GroupAssigneeEntity
+    from snuba.datasets.cdc.groupedmessage_entity import GroupedMessageEntity
+    from snuba.datasets.entities.discover import DiscoverEntity
+    from snuba.datasets.entities.errors import ErrorsEntity
+    from snuba.datasets.entities.events import EventsEntity
+    from snuba.datasets.entities.groups import GroupsEntity
+    from snuba.datasets.entities.outcomes import OutcomesEntity
+    from snuba.datasets.entities.outcomes_raw import OutcomesRawEntity
+    from snuba.datasets.entities.sessions import SessionsEntity
+    from snuba.datasets.entities.transactions import TransactionsEntity
+
+    entity_factories: MutableMapping[str, Callable[[], Entity]] = {
```

Comment:
The type hint is wrong. The keys of this mapping are not strings. They are EntityKey